iina/Extensions.swift

```diff
@@ -227,6 +227,23 @@ extension Double {
       return "\(rounded)"
     }
   }
+
+  func roundedTo2Decimals() -> Double {
+    let scaledUp = self * 1e2
+    let scaledUpRounded = scaledUp.rounded(.up)
+    let finalVal = scaledUpRounded / 1e2
+    return finalVal
+  }
+  
+  /// Formats this number as a decimal string, using default locale.
+  ///
+  /// This should be used in most places where decimal numbers need to be printed.
+  var string: String {
+    /// Output up to 15 digits after the decimal. This should be good enough for almost all cases, but can be increased in the future.
+    /// (It's not clear what the maximum allowable value for `NumberFormatter.maximumFractionDigits` actually is. Trying to use
+    /// `NSIntegerMax` seems to result in `maximumFractionDigits` being silently set to `6` instead.)
```

Comment:
I believe markdown comments can only be used on declarations. I believe comments inside code blocks should just be simple (//) comments. I tried finding Swift documentation that explains where markdown comments can be used and struck out. I don't think there is any way to get Xcode to show this comment in the quick help. It will show the comment on the declaration if you click on `string`.

For small blocks of code like this I will put the fancy comment outside of the code and in the comment on the declaration.

Reply:
I tend to default to the "fancy comments" because of their ability to highlight `code snippits` in them. It's also annoying to start with a "regular" double-slash comment and then have to change it. But in this case I think you are right to point out that I should have just put these details above the declaration because they could be important. Fixed in latest.